http_proxy.go

```diff
@@ -1016,15 +1018,31 @@ func (p *Proxy) listenWATER(addr string) (net.Listener, error) {
 		wasm = wasmBuffer.Bytes()
 	}
 
-	// currently the WATER listener doesn't accept a multiplexed connections, so we need to listen and accept connections directly from the listener
-	waterListener, err := water.NewWATERListener(ctx, nil, p.WaterTransport, addr, wasm)
+	mismatchProtocol, err := strconv.Atoi(p.WaterMismatchProtocol)
 	if err != nil {
-		log.Errorf("failed to starte WATER listener: %w", err)
-		return nil, err
+		return nil, log.Errorf("failed to parse mismatch protocol")
 	}
 
-	log.Debugf("Listening for water at %v", waterListener.Addr())
-	return waterListener, nil
+	switch mismatchProtocol {
+	case 0:
+		// currently the WATER listener doesn't accept a multiplexed connections, so we need to listen and accept connections directly from the listener
+		waterListener, err := water.NewWATERListener(ctx, nil, p.WaterTransport, addr, wasm)
+		if err != nil {
+			return nil, log.Errorf("failed to starte WATER listener: %w", err)
+		}
+
+		log.Debugf("Listening for water at %v", waterListener.Addr())
+		return waterListener, nil
+	case 97:
+		cert, err := utls.X509KeyPair([]byte(p.CertFile), []byte(p.KeyFile))
+		if err != nil {
+			return nil, log.Errorf("failed to load cert: %w", err)
+		}
+
+		return utls.Listen("tcp", addr, &utls.Config{Certificates: []utls.Certificate{cert}})
```

Comment:
You probably don't need `utls` for this FYI.  `utls` should be compatible with any TLS server, including the one offered by `crypto/tls`.  It won't hurt to use `utls` here, but using `crypto/tls` would make it more clear to maintainers that we're not doing anything fancy on the server side. Using `crypto/tls` would also be slightly safer as that library gets security updates more frequently than `utls` (which just periodically merges in changes to `crypto/tls`).

Reply:
To expand on this, just for informational purposes, `utls` on the client is generally just used to modify the ClientHello (the first message sent by TLS clients).  The ClientHellos produced by `utls` are still legal by the TLS spec, so any TLS server should be able to handle them.  On the server side, you would use `utls` to inspect or modify details of the connection at a level not offered by `crypto/tls`.  Again, it would all be legal by the TLS spec AFAIK.  Since we're not doing anything fancy with `utls` here, we could just use `crypto/tls`.

Reply:
Aaaaaaaaah I was afraid there was any incompatibility between those implementations, I'll replace this by the crypto/tls

Reply:
Just for documentation purposes: I've tested this implementation with `utls` and it worked successfully! Tomorrow I should move to `crypto/tls`, test again and this ticket should be ready for review.